runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -270,16 +267,47 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		}
 		else if (menuAction == MenuAction.EXAMINE_NPC && client.isKeyPressed(KeyCode.KC_SHIFT))
 		{
-			// Add tag option
+			// Add tag and tag-all options
+			final int id = event.getIdentifier();
+			final NPC[] cachedNPCs = client.getCachedNPCs();
+			final NPC npcCached = cachedNPCs[id];
+
+			if (npcCached == null || npcCached.getName() == null)
+			{
+				return;
+			}
+
+			final String npcName = npcCached.getName().toLowerCase();
+			boolean	matchesList = highlights.stream()
```

Comment:
This should include a comment explaining what it is trying to do, eg
```java
// Only add Untag options to npcs not highlighted by a wildcard entry, because untag-all will not remove wildcards
```